packages/@react-aria/slider/src/useSlider.ts

```diff
@@ -61,46 +61,48 @@ export function useSlider<T extends number | number[]>(
   let {addGlobalListener, removeGlobalListener} = useGlobalListeners();
 
   // When the user clicks or drags the track, we want the motion to set and drag the
-  // closest thumb.  Hence we also need to install useMove() on the track element.
+  // closest thumb. Hence, we also need to install useMove() on the track element.
   // Here, we keep track of which index is the "closest" to the drag start point.
   // It is set onMouseDown/onTouchDown; see trackProps below.
   const realTimeTrackDraggingIndex = useRef<number | null>(null);
 
-  const stateRef = useRef<SliderState>(null);
-  stateRef.current = state;
+
   const reverseX = direction === 'rtl';
   const currentPosition = useRef<number>(null);
-  const {moveProps} = useMove({
-    onMoveStart() {
-      currentPosition.current = null;
-    },
-    onMove({deltaX, deltaY}) {
-      let {height, width} = trackRef.current.getBoundingClientRect();
-      let size = isVertical ? height : width;
+  let onMoveStart = useEffectEvent(() => {
+    currentPosition.current = null;
+  });
+  let onMove = useEffectEvent(({deltaX, deltaY}) => {
+    let {height, width} = trackRef.current.getBoundingClientRect();
+    let size = isVertical ? height : width;
 
-      if (currentPosition.current == null) {
-        currentPosition.current = stateRef.current.getThumbPercent(realTimeTrackDraggingIndex.current) * size;
-      }
+    if (currentPosition.current == null) {
+      currentPosition.current = state.getThumbPercent(realTimeTrackDraggingIndex.current) * size;
+    }
 
-      let delta = isVertical ? deltaY : deltaX;
-      if (isVertical || reverseX) {
-        delta = -delta;
-      }
+    let delta = isVertical ? deltaY : deltaX;
+    if (isVertical || reverseX) {
+      delta = -delta;
+    }
 
-      currentPosition.current += delta;
+    currentPosition.current += delta;
 
-      if (realTimeTrackDraggingIndex.current != null && trackRef.current) {
-        const percent = clamp(currentPosition.current / size, 0, 1);
-        stateRef.current.setThumbPercent(realTimeTrackDraggingIndex.current, percent);
-      }
-    },
-    onMoveEnd() {
-      if (realTimeTrackDraggingIndex.current != null) {
-        stateRef.current.setThumbDragging(realTimeTrackDraggingIndex.current, false);
-        realTimeTrackDraggingIndex.current = null;
-      }
+    if (realTimeTrackDraggingIndex.current != null && trackRef.current) {
+      const percent = clamp(currentPosition.current / size, 0, 1);
+      state.setThumbPercent(realTimeTrackDraggingIndex.current, percent);
     }
   });
+  let onMoveEnd = useEffectEvent(() => {
+    if (realTimeTrackDraggingIndex.current != null) {
+      state.setThumbDragging(realTimeTrackDraggingIndex.current, false);
+      realTimeTrackDraggingIndex.current = null;
+    }
+  });
```

Comment:
I think these `useEffectEvent`s should really be inside `useMove` so that other consumers benefit as well. I made this change in #4564.